hibernate-core/src/main/java/org/hibernate/proxy/pojo/javassist/JavassistProxyFactory.java

```diff
@@ -93,7 +93,21 @@ public void postInstantiate(
 		javassist.util.proxy.ProxyFactory factory = new javassist.util.proxy.ProxyFactory() {
 			@Override
 			protected ClassLoader getClassLoader() {
-				return persistentClass.getClassLoader();
+				// This classloader does not take into account class space consistency
+				// i.e. if the persistenceClass can see the same interface types via
+				// a different ClassLoader hierarchy. In that case we could try some 
+				// risky magic with loadClass, but using findClass is much safer.
+				return new ClassLoader(persistentClass.getClassLoader()) {
```

Comment:
Thanks for the explanation, that helps.  So I think access ClassLoaderService is the best option.

BUT

I am a little nervous of pushing a set of ClassLoaders to a static variable on a class that is very likely loaded by the container/environment.